src/types/global.d.ts

```diff
@@ -32,4 +32,6 @@ declare module '*.lottie' {
 interface Window {
     enableMemoryOnlyKeys: () => void;
     disableMemoryOnlyKeys: () => void;
+    setSupportToken: (token: string, email: string, accountID: number) => void;
+    Onyx: Record<string, unknown>;
 }
```

Comment:
Instead of Record<string, unknown> let's use the real type, something like this should work?

```suggestion
import Onyx from 'react-native-onyx';

interface Window {
    enableMemoryOnlyKeys: () => void;
    disableMemoryOnlyKeys: () => void;
    setSupportToken: (token: string, email: string, accountID: number) => void;
    Onyx: Onyx;
}
```

Reply:
Please research it a bit and try different approaches 😄 (if you fail, let me know and I'll try to help)

Reply:
@fabioh8010 Any ideas?

Reply:
My proposal to make it work:

1) Change `src/types/global.d.ts` to have only this in `Window`.

```ts
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
interface Window {
    setSupportToken: (token: string, email: string, accountID: number) => void;
}
```

2) Change `src/types/modules/react-native-onyx.d.ts` to include the remaining methods (they are related to Onyx so I think it makes sense to have them there)

```ts
import type Onyx from 'react-native-onyx';
import type {OnyxCollectionKey, OnyxKey, OnyxValues} from '@src/ONYXKEYS';

declare module 'react-native-onyx' {
    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
    interface CustomTypeOptions {
        keys: OnyxKey;
        collectionKeys: OnyxCollectionKey;
        values: OnyxValues;
    }
}

declare global {
    // Global methods for Onyx key management for debugging purposes
    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
    interface Window {
        enableMemoryOnlyKeys: () => void;
        disableMemoryOnlyKeys: () => void;
        Onyx: typeof Onyx;
    }
}
```

I tested here, ran `npm run typecheck` and everything looks okay, wdyt? @mkhutornyi @blazejkustra

Reply:
@mkhutornyi bump